modules/browsemap/src/test/java/com/sygic/modules/browsemap/BrowseMapFragmentViewModelTest.kt

```diff
@@ -0,0 +1,310 @@
+            themeManager
+        )
+    }
+
+    @Test
+    fun initTest() {
+        verify(mapInteractionManager).addOnMapClickListener(browseMapFragmentViewModel)
+    }
+
+    @Test
+    fun initComponentTest() {
+        assertEquals(browseMapFragmentViewModel.mapSelectionMode, MapSelectionMode.FULL)
+        assertEquals(browseMapFragmentViewModel.positionOnMapEnabled, true)
+        assertEquals(browseMapFragmentViewModel.compassEnabled.value, true)
+        assertEquals(browseMapFragmentViewModel.compassHideIfNorthUp.value, true)
+        assertEquals(browseMapFragmentViewModel.positionLockFabEnabled.value, true)
+        assertEquals(browseMapFragmentViewModel.zoomControlsEnabled.value, true)
+
+        verify(locationManager).requestToEnableGps(any(), any())
+        verify(permissionsManager).checkPermissionGranted(anyString(), any())
+    }
+
+    @Test
+    fun onStartTestPositionOnMapEnabled() {
+        whenever(locationManager.positionOnMapEnabled).thenReturn(true)
+        browseMapFragmentViewModel.onStart(mock())
+        verify(locationManager).setSdkPositionUpdatingEnabled(true)
+    }
+
+    @Test
+    fun onStartTestPositionOnMapDisabled() {
+        whenever(locationManager.positionOnMapEnabled).thenReturn(false)
+        browseMapFragmentViewModel.onStart(mock())
+        verify(locationManager, never()).setSdkPositionUpdatingEnabled(true)
+    }
+
+    @Test
+    fun onMapObjectsRequestStartedTest() {
+        browseMapFragmentViewModel.onMapObjectsRequestStarted()
+        verify(extendedMapDataModel).removeOnClickMapMarker()
+    }
+
+    @Test
+    fun onMapObjectsReceivedEmptyListTest() {
+        browseMapFragmentViewModel.onMapObjectsReceived(listOf())
+        verify(poiDataManager, never()).getViewObjectData(any(), any())
+    }
+
+    @Test
+    fun onMapObjectsReceivedSelectionModeNoneTest() {
+        val testMapMarker = MapMarker.at(48.143489, 17.150560).build()
+
+        browseMapFragmentViewModel.mapSelectionMode = MapSelectionMode.NONE
+        browseMapFragmentViewModel.onMapObjectsReceived(listOf(testMapMarker, mock()))
+        verify(poiDataManager, never()).getViewObjectData(any(), any())
+    }
+
+    @Test
+    fun onMapObjectsReceivedSelectionModeMarkersOnlyCirclesListTest() {
+        val tesViewObject = MapCircle(GeoCoordinates(48.143489, 17.150560), 60.0)
+
+        browseMapFragmentViewModel.mapSelectionMode = MapSelectionMode.MARKERS_ONLY
+        browseMapFragmentViewModel.onMapObjectsReceived(listOf(tesViewObject))
+        verify(poiDataManager, never()).getViewObjectData(any(), any())
+    }
+
+    @Test
+    fun onMapObjectsReceivedSelectionModeMarkersOnlyMapMarkerListTest() {
+        val testMarkerData = MarkerData(48.143489, 17.150560, BasicData("Test"))
+        val testMapMarker = MapMarker.from(testMarkerData).build()
+
+        argumentCaptor<PoiDataManager.Callback>().let { callback ->
+            whenever(poiDataManager.getViewObjectData(any(), callback.capture())).then {
```

Comment:
🤔 to ze to ten let odsadzuje tak posobi, ze je to nejaky callback alebo nieco, ze to `whenever` sa zavola az ako dosledok. Avsak ono je to vlastne na rovnakej urovni ako vsetky ostatne volania a vykona sa zakazdym

nebolo by to predsa prehladnejsie takto?

```
val callback = argumentCaptor<PoiDataManager.Callback>()
whenever(poiDataManager.getViewObjectData(any(), callback.capture())).then {
    callback.firstValue.onDataLoaded(testMarkerData)
}
```

Reply:
možme klidně tak ;)